### ISSUE TITLE
Ethernet connection as a run condition

### ISSUE THREAD

**Issue by mikulik86** [Catfriend1/syncthing-android#899]
Hello and thanks to the developer. 

Could you please add a ethernet connection as another run condition?
Would be great.

**Comment by Catfriend1** [comment]
What is the use case why this run condition is so important for you?

**Comment by mikulik86** [comment]
I put my phone over night into a flight mode and connect it to a dock for charging and ethernet connectivity. And I want it to sync only then. Not other times when the phone is charging somewhere else where also connected to a Wi-Fi.

**Comment by Catfriend1** [comment]
I think the run condition "run in flight mode" could help you out in this scenario.

If you'd like to implement a new run condition, that would be mainly copy and paste in run condition monitor and the layout of the settings screen xml file plus some strings.

**Comment by mikulik86** [comment]
> I think the run condition "run in flight mode" could help you out in this scenario.

Not really. It is possible to have flight mode on and be on Wi-Fi simultaneously, which would trigger unwanted syncing.
Do you think you could implement the ethernet condition?

**Comment by Catfriend1** [comment]
Yes, I wrote how it should be done above. And no, I'm not motivated to take my time here.

**Comment by mikulik86** [comment]
Unfortunately my extremely basic programming knowledge is not enough to know what exactly your instructions mean and how to do it. 
It's a pity, nonetheless thanks for this otherwise great fork.

**Comment by martin-braun** [comment]
@Catfriend1 I'm not interested in the feature here, but maybe you should give people the opportunity to sponsor you through GitHub Sponsor for nice badges, etc. That could rise your motivation, even maintaining this is effort.

**Comment by martin-braun** [comment]
Would this also affect USB Tethering? In this case I am indeed interested in this feature.

**Comment by Catfriend1** [comment]
Maybe it can be achieved by third party automation tools, e.g. https://llamalab.com/automate/community/flows/41583

**Comment by mikulik86** [comment]
> Maybe it can be achieved by third party automation tools, e.g. https://llamalab.com/automate/community/flows/41583

Thank you, this works nice for me. 
There is just one minor disadvantage: when controlled by Broadcast, force start/stop doesn't work. I suppose you wouldn't want to change it, would you?

**Comment by Catfriend1** [comment]
I could look into it. Or you suggest a PR to make the buttons available when controlled by broadcast?

**Comment by mikulik86** [comment]
I would be glad if you could look into it and make your software even better.
Thanks in advance.

**Comment by Konfekt** [comment]
While @mikulik86 got it working with

> https://llamalab.com/automate/community/flows/41583

for Tasker, the following worked for me to sync it with the computer when connected to it via USB (or Wifi, which was already available as an option inside Syncthing Fork):

1. In Tasker, add 
	- two state entry tasks, one for Wifi and one for Power connect (USB or Ether didn't trigger), both triggering an Intent using package `com.github.catfriend1.syncthingandroid` and Broadcast  `com.github.catfriend1.syncthingandroid.action.START`
	- a single state entry task for simultaneous Wifi and Power disconnect, triggering an Intent using package `com.github.catfriend1.syncthingandroid` and Broadcast  `com.github.catfriend1.syncthingandroid.action.STOP`
2. In Syncthing Fork, enable the experimental setting Listen to Broadcast .
3. On the computer, make sure to have permit Syncthing's ports for the USB connection it uses

Maybe these workarounds for those syncing via other connections than WWAN/WLAN could be added to the Wiki ? It seems write locked.

**Comment by Catfriend1** [comment]
I am currently not into topics regarding the run condition engine as I am not using it myself and troubleshooting a permanent service in an Android ecosystem where Google's policies don't want that behaviour is cat and mouse fight. My focus lies on the main wrapper function, atm.
